app/controllers/events_controller.rb

```diff
@@ -0,0 +1,79 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authenticate_admin!, only: [:edit, :update, :destroy]
+  before_action :authenticate_user!, only: [:index, :new]
+
+  # GET /events
+  # GET /events.json
+  def index
+    @events = Event.all
+  end
+
+  # GET /events/1
+  # GET /events/1.json
+  def show
+  end
+
+  # GET /events/new
+  def new
+    @event = Event.new
+  end
+
+  # GET /events/1/edit
+  def edit
+      #redirect_to modify_event_from_admin_form(params)
+  end
+
+  # POST /events
+  # POST /events.json
+  def create
+    @event = Event.new(event_params)
+
+    respond_to do |format|
+      if @event.save
+        flash['success'] = t('events.new.event_create_success')
```

Comment:
Can you use lazy loading here?

https://stackoverflow.com/questions/33391908/how-to-use-i18n-from-controller-in-rails